crates/async-dropper-simple/Cargo.toml

```diff
@@ -14,8 +14,8 @@ A simple struct-wrapper (i.e. AsyncDropper<T>) based implementation of AsyncDrop
 
 [features]
 default = []
-tokio = ["dep:tokio"]
-async-std = ["dep:async-std"]
+tokio = ["dep:tokio", "dep:async-scoped", "async-scoped/use-tokio"]
+async-std = ["dep:async-std",  "dep:async-scoped", "async-scoped/use-async-std"]
```

Comment:
What do you think about actually making a new feature flag for this? 

We can avoid the breaking change, allow people to opt in to using non-`'static` lifetimes and the new functionality here.

Reply:
This is not a breaking change, those that want's to spawn static can still do so.

Reply:
I think I misunderstood your other comment about "breaking", I thought you were suggesting that the non non-`'static` was the breaking part. 

That said, wouldn't the increased derive requirements:

```rust
#[derive(Debug, PartialEq, Eq, Hash, Clone)]
pub struct AsyncDropper<T: AsyncDrop + Send> {
```

Be a breaking change for anyone who has a `T` which doesn't already implement these but only implements `Default` since up until now we only used `Default`?

Reply:
To clarify why 'static was removed:
A normal drop from any function doesn't really require 'static, it's just cleaning up it's scope, and 'static is not it's scope, it should be perfectly fine operating on it's current scope and not require it to be 'static. In matter of fact it's incorrect to require it, it's like saying the local function cannot mutate it's own state.

Reply:
Thanks for the additional explanation -- Yeah I see that the `async-scoped` addition enables the removal of `'static` -- IIRC the `'static` was required only because of the usual bounds required on async functions. I didn't know about `async-scoped` so I didn't know there was a reasonable way to make that bound unnecessary.

Separate from that, as far as the huge structs, usually a huge struct is an anti pattern AFAIK, and I do wonder if it's dramatically easier to implement `Default` versus `Debug, PartialEq, Eq, Hash, Clone`, for a struct that perhaps didn't implement those before.

Reply:
It only tries to implement those if T has it.

Reply:
The reason is to easier store AsyncDropper inside another struct, that really needed Clone, PartialEq, Hash and so on.

Reply:
Yeah I did a [little playground example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=8330486db03655b2996a6149a152faa9) to test that out -- it only becomes a problem if you try to use one of those bounds and the inner `T` doesn't implement it. 

Again, thanks for the extra explanation